[CI] Update nightly builds to use JDK 22

jerboaa opened the pull request:
This PR does the following things:

- Update nightly builds to only build mandrel with JDK 22. It has a higher chance of working over JDK 21 due to #598
- Keep building `graal-source` in the JDK 21 mode
- Also build `graal-source` in the JDK 22 config

jerboaa committed:
[CI] Update nightly builds to use JDK 22

zakkak commented on .github/workflows/nightly.yml:
```diff
@@ -40,34 +51,34 @@ jobs:
       build-type: "graal-source"
       jdk: "21/ea"
-      build-stats-tag: "gha-linux-qmain-glatest-jdk21ea"
+      build-stats-tag: "gha-linux-graal-qmain-glatest-jdk21ea"
```
FWIW `glatest` stands for graal latest and `mlatest` for mandrel latest. I know, not the best choice... :)

jerboaa replied:
aah, yeah hard to spot.

zakkak commented:
> * Update nightly builds to only build mandrel with JDK 22. It has a higher chance of working over JDK 21 due to https://github.com/graalvm/mandrel/issues/598

Since mandrel with JDK 22 is also tested on the weekly builds, can you please update them to build with JDK 21 instead? 

> * Keep building graal-source in the JDK 21 mode
> * Also build graal-source in the JDK 22 config

I suggest we remove the JDK 21 build (or move it to the weekly job), graal builds from source with JDK 21 are being tested nightly in https://github.com/oracle/graal/actions/workflows/quarkus.yml as well

jerboaa committed:
Disable JDK 22 mandrel testing in weekly

We test it nightly and using JDK 21 is bound to fail due to
graalvm/mandrel/issue/598.

jerboaa committed:
Move GraalVM CE JDK 21 build to weekly

jerboaa commented:
> > * Update nightly builds to only build mandrel with JDK 22. It has a higher chance of working over JDK 21 due to [GraalVM master fails to build with OpenJDK 21 after GR-49610 #598](https://github.com/graalvm/mandrel/issues/598)
> 
> Since mandrel with JDK 22 is also tested on the weekly builds, can you please update them to build with JDK 21 instead?

I've disabled JDK 22 builds in the weekly job for now. JDK 21-based mandrel builds are a moot point currently due to #598.
 
> > * Keep building graal-source in the JDK 21 mode
> > * Also build graal-source in the JDK 22 config
> 
> I suggest we remove the JDK 21 build (or move it to the weekly job), graal builds from source with JDK 21 are being tested nightly in https://github.com/oracle/graal/actions/workflows/quarkus.yml as well

OK, since I don't monitor graal jobs upstream, I prefer having it in the mandrel actions. Moved to the weekly job, which should be sufficient.

zakkak commented:
> JDK 21-based mandrel builds are a moot point currently due to https://github.com/graalvm/mandrel/issues/598.

OK, so you suggest we keep them disabled till we have a fix.

jerboaa commented:
> > JDK 21-based mandrel builds are a moot point currently due to #598.
> 
> OK, so you suggest we keep them disabled till we have a fix.

Yes. No point in having known failing CI. It's just noise.
